### PR TITLE
DEPR: **kwargs in ExcelWriter

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Currently ExcelWriter has **engine_kwargs to pass through to the engine, [although this is undocumented](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.ExcelWriter.html). All other places in pandas where engine_kwargs is used, they are a regular argument.

This PR 

 - Renames **engine_kwargs to **kwargs (not an API change)
 - Adds **kwargs to the docstring of ExcelWriter
 - Deprecates **kwargs
 - Introduces engine_kwargs as a regular argument for their replacement

### CODE REVIEW

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -667,6 +667,14 @@ class ExcelWriter(metaclass=abc.ABCMeta):
         be parsed by ``fsspec``, e.g., starting "s3://", "gcs://".
 
         .. versionadded:: 1.2.0
+    engine_kwargs : dict, optional
+        Keyword arguments to be passed into the engine.
```
Comment (jreback) [inline pandas/io/excel/_base.py:671]: versionadded 1.3

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -745,7 +753,26 @@ class ExcelWriter(metaclass=abc.ABCMeta):
...
+        **kwargs,
+    ):
+        if kwargs:
+            if engine_kwargs is not None:
+                raise ValueError("Cannot use both engine_kwargs and **kwargs")
+            warnings.warn(
+                "Use of **kwargs is deprecated, use engine_kwargs instead.",
+                FutureWarning,
+                stacklevel=2,
+            )
```
Comment (jreback) [inline pandas/io/excel/_base.py:774]: shouldn't you simply do `engine_kwargs = kwargs` here?

File: pandas/io/excel/_base.py [hunk 3]
```diff
@@ -835,7 +862,8 @@ def __init__(
         datetime_format=None,
         mode: str = "w",
         storage_options: StorageOptions = None,
-        **engine_kwargs,
+        engine_kwargs: Optional[Dict] = None,
+        **kwargs,
```
Comment (jreback) [inline pandas/io/excel/_base.py:866]: and not do this

File: pandas/io/excel/_base.py [hunk 4]
```diff
@@ -745,7 +753,26 @@ class ExcelWriter(metaclass=abc.ABCMeta):
...
+        date_format=None,
+        datetime_format=None,
+        mode: str = "w",
+        storage_options: StorageOptions = None,
+        engine_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        if kwargs:
+            if engine_kwargs is not None:
+                raise ValueError("Cannot use both engine_kwargs and **kwargs")
```
Comment (jreback) [inline pandas/io/excel/_base.py:769]: can you add a test for this

File: pandas/io/excel/_base.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/io/excel/_base.py:774]: Both engine_kwargs and kwargs go entirely unused in `_base.ExcelWriter.__new__` and `_base.ExcelWriter.__init__`; they are only utilized in the subclasses for each engine.

**Comment by rhshadrach** [comment]
Thanks @jreback - changes made and green. I might not have understood your comment about setting `engine_kwargs = kwargs`, response is above.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**